src/Helpers.jl

```diff
@@ -34,3 +34,11 @@ function Base.iterate(it::RingPeek{Itr}, state) where {Itr}
 
     ((x, peeked), newstate)
 end
+
+function get_permutator(permutator)
```

Comment:
Rename it to a better name, because it can be useful in other contexts.
Also, this function modifies the `permutator` variable **in-place**, so the convention is to suffix it with `!`.
```suggestion
function normalizeperm!(permutator)
```